pkg/core/blockchain_test.go

```diff
@@ -1892,13 +1892,13 @@ func benchmarkForEachNEP17Transfer(t *testing.B, ps storage.Store, nBlocks int)
 
 	lastB, err := bc.GetBlock(bc.GetHeaderHash(int(bc.BlockHeight())))
 	require.NoError(t, err)
-	_ = lastB.Timestamp
+	newestTimestamp := lastB.Timestamp
 
 	t.ResetTimer()
 	t.ReportAllocs()
 	t.StartTimer()
 	for i := 0; i < t.N; i++ {
-		bc.ForEachNEP17Transfer(priv0ScriptHash, func(transfer *state.NEP17Transfer) (bool, error) {
+		bc.ForEachNEP17Transfer(priv0ScriptHash, newestTimestamp, func(transfer *state.NEP17Transfer) (bool, error) {
```

Comment:
You're looping through the whole set starting from the latest block here and for this case there won't be any meaningful difference. Try emulate paging though transfers (get 100 latest, then 100 before them, then ...) and you're likely to see that the deeper you get the more effective this optimization is.

Reply:
It's done intentionally. IMO, If we're emulating paging, then it's kind of a "cheating" benchmark, because the old implementation has to explicitly list through all starting pages to get the needed one, and the new implementation starts right from the needed page. It's obvious that the new implementation wins in those cases when we're starting far away from the current chain's height. And what I'd really like to see is how the new implementation behaves in case of equal conditions, i.e. starting from the newest page, because it's default `getnep17balances` behaviour. So even in this edge case and even iterating over the whole set of transfers the new implementation wins on long distances.

But I agree, iterating through the whole set of transfers is an artificial case, so I refactored this benchmark. Now it behaves more like our RPC paging.

Reply:
I'll attach the new benchmark results as soon as it's finished.

Reply:
> it's kind of a "cheating" benchmark

I wouldn't say it's cheating. For the first page it's supposed to be about the same, then with every page the new implementation is supposed to win more and more. Now we need to see whether that's the case.

Reply:
Yes, it is, see the updated commit message.